gravitee-io/gravitee-access-management
# fix: check deployed status of a plugin before creating the configuration

fixes AM-732

## description

This PR aims to reject a EE plugin creation if the plugin is not deployed (plugin not part of the license or missing from the installation)
Check apply on:
* IdentityProvider
* BotDetection
* Factor
* Resources
* Policies
* Reporter
* ExtensionGrantType

## How to test

On EE environment
* Open network view of the brower
* create EE IDP
* copy the curl command from the network view
* do the same for EE factor, EE resource, and a EE plugin.

On CE environmen:
* use the curl commands to have the right payloas (adapt the internal ids)
* the request should fail with the message '`Plugin type xxx not deployed`'
